src/main/kotlin/io/github/mojira/arisa/domain/Link.kt

```diff
@@ -1,10 +1,8 @@
 package io.github.mojira.arisa.domain
 
-import arrow.core.Either
-
 data class Link(
     val type: String,
     val outwards: Boolean,
     val issue: LinkedIssue,
-    val remove: () -> Either<Throwable, Unit>
```

Comment:
We are saying this cant fail anymore? I dont think thats correct?
Same for all other functions

Reply:
All API requests are sent at the end of each run; what the `remove` function here does is to add the remove operation to the update context's `otherOperations` list, which will never fail

Reply:
what I mean is that at the end of the run it might fail, and if it does we might not want to have done other tasks (like commenting if it failed resolving, as it will just start spamming comments). Thats where a promise might help. Its a operation that will be executed async at the end of the run but might still fail so we might want to execute things if it fails, things if it doesnt

Reply:
Ah I think I get you now. Then what should we do when it fails so that we can prevent it from spamming comments in your example?

Reply:
It requires a change on how we are doing things now, but:

1- We make it do all actions in the same module, so multiple modules have multiple caches
2- The functions return a promise, and we combine the multiple actions as multiple promises. Then we do a .onError {} after they executed to do one action(if the module requires), flatMap to do something if everything else is succesful(where we might do more jira request that dont go through the cache)

Honestly 1 is simpler to implement, but it would cause multiple changes, can we calculate how many changes in both options we save in common cases?

Reply:
The first option can already resolve the multiple changes problem caused by the `TransferVersionModule` mentioned in #192.

The second option can combine changes to fields like `Linked` and `Confirmation Status` furthermore, which I honestly don't think worth the work ;P

And if we decide to take the first option, how do you feel about moving the applying-changes-and-clearing-cache part to the `executeModule` function?

Reply:
it would be better, yes :D